specs/ClientCertificate.md

```diff
@@ -0,0 +1,500 @@
+} COREWEBVIEW2_CLIENT_CERTIFICATE_KIND;
+
+[uuid(92d41f20-8e92-11eb-8dcd-0242ac130003), object, pointer_default(unique)]
+interface ICoreWebView2_3 : ICoreWebView2_2 {
+  /// Add an event handler for the ClientCertificateRequested event.
+  /// The ClientCertificateRequested event is raised when the WebView2
+  /// is making a request to an HTTP server that needs a client certificate
+  /// for HTTP authentication and when there is atleast one mutually trusted certitifcate.
+  /// Read more about HTTP client certificates at
+  /// [RFC 8446 The Transport Layer Security (TLS) Protocol Version 1.3](https://tools.ietf.org/html/rfc8446).
+  ///
+  /// With this event you have several options for responding to client certificate requests:
+  ///
+  /// *  You can query the mutually trusted CA certificates, and select a certificate
+  /// from the list to respond to the server.
+  /// *  You can choose to respond to the server without a certificate.
+  /// *  You can choose to display default client certificate selection dialog prompt
+  /// to let user to respond to the server.
+  /// *  You can cancel the request.
+  ///
+  /// If you doesn't handle the event, WebView2 will
+  /// show the default client certificate selection dialog prompt to user.
+  ///
+  /// \snippet SettingsComponent.cpp ClientCertificateRequested
+  HRESULT add_ClientCertificateRequested(
+      [in] ICoreWebView2ClientCertificateRequestedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove an event handler previously added with add_ClientCertificateRequested.
+  HRESULT remove_ClientCertificateRequested([in] EventRegistrationToken token);
+}
+
+/// Provides access to the certificate metadata
+[uuid(9ca02116-8e92-11eb-8dcd-0242ac130003), object, pointer_default(unique)]
+interface ICoreWebView2ClientCertificate : IUnknown {
+  /// Subject of the certificate.
+  [propget] HRESULT Subject([out, retval] LPWSTR* subject);
+  /// Name of the certificate authority that issued the certificate.
+  [propget] HRESULT Issuer([ out, retval ] LPWSTR* issuer);
+  /// The valid start date and time for the certificate as the number of seconds since
+  /// the UNIX epoch.
+  [propget] HRESULT ValidStartDate([out, retval] double* validStartDate);
+  /// The valid expiration date and time for the certificate as the number of seconds since
+  /// the UNIX epoch.
+  [propget] HRESULT ValidExpiryDate([out, retval] double* validExpiryDate);
+  /// DER encoded serial number of the certificate.
+  [propget] HRESULT SerialNumber([out, retval] LPWSTR* serialNumber);
+  /// Display name for a certificate.
+  [propget] HRESULT DisplayName([out, retval] LPWSTR* displayName);
+  /// PEM encoded data for the certificate.
+  /// Returns Base64 encoding of DER encoded certificate.
+  /// Read more about PEM at [RFC 1421 Privacy Enhanced Mail]
+  /// (https://tools.ietf.org/html/rfc1421).
+  [propget] HRESULT PEMEncodedData([out, retval] LPWSTR* pemEncodedData);
```

Comment:
Windows Runtime convention is to use word-case for acronyms of 3 letters or more, so PemEncodedData.

The name and description of this property suggests that a certificate has a "data" payload of some kind, and this property returns the data (in PEM encoding, presumably not encapsulated; just raw data).

But I think what this actually does is return the certificate itself, in PEM encoding. Would this be a method Certificate.ToPemEncoding(), similar to how we use ToJson for other types?

On the other hand, this pattern doesn't extend easily to the next property, which returns an array of PEM-encoded certificates.

Reply:
The ToPemEncoding name makes it much clearer what this method does (assuming that's what it does). Please rename if so.